src/sync.h

```diff
@@ -97,21 +100,23 @@ class CCriticalSection : public AnnotatedMixin<boost::recursive_mutex>
 };
 
 /** Wrapped boost mutex: supports waiting but not recursive locking */
-typedef AnnotatedMixin<boost::mutex> CWaitableCriticalSection;
+typedef AnnotatedMixin<std::mutex> CWaitableCriticalSection;
 
-/** Just a typedef for boost::condition_variable, can be wrapped later if desired */
-typedef boost::condition_variable CConditionVariable;
+/** Lock for CConditionVariable */
+typedef std::unique_lock<std::mutex> CWaitableLock;
+
+/** Just a typedef for std::condition_variable, can be wrapped later if desired */
+typedef std::condition_variable CConditionVariable;
 
 #ifdef DEBUG_LOCKCONTENTION
 void PrintLockContention(const char* pszName, const char* pszFile, int nLine);
 #endif
 
 /** Wrapper around boost::unique_lock<Mutex> */
-template <typename Mutex>
-class SCOPED_LOCKABLE CMutexLock
+class SCOPED_LOCKABLE CCriticalBlock
```

Comment:
In commit "Fix -Wthread-safety-analysis warnings."

The naming here doesn't seem very consistent:

Recursive mutex: **CCriticalSection** (unchanged)
Recursive lock: **CCriticalBlock** (previously _CMutexLock_)
Non-recursive mutex: **CWaitableCriticalSection** (unchanged)
Non-recursive lock: **CWaitableLock** (previously didn't exist)

I don't see why we wouldn't just follow the standard naming and our current coding convention and use:

Recursive mutex: **RecursiveMutex**
Recursive lock: **RecursiveLock**
Non-recursive mutex: **Mutex**
Non-recursive lock: **Lock**

Maybe defining backwards compatible _CCriticalSection_ and _CWaitableCriticalSection_ typedefs to avoid increasing the size of the diff.